src/thingset_priv.h

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2017 Martin Jäger / Libre Solar
+ * Copyright (c) 2021 Bobby Noelte.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef THINGSET_PRIV_H_
+#define THINGSET_PRIV_H_
+
+#include "thingset.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef __ZEPHYR__
+
+/* Logging */
+#define LOG_MODULE_NAME thingset
+#define LOG_LEVEL CONFIG_THINGSET_LOG_LEVEL
+#include <logging/log.h>
+
+#ifdef THINGSET_MAIN
+LOG_MODULE_REGISTER(LOG_MODULE_NAME);
+#else
+LOG_MODULE_DECLARE(LOG_MODULE_NAME);
+#endif
+
+#define LOG_ALLOC_STR(str)	((str == NULL) ? log_strdup("null") : \
+                                                log_strdup(str))
+
+#include <zephyr.h>
+
+#else /* ! __ZEPHYR */
+
+#define DEBUG 0
+
+#define LOG_DBG(...) printf(__VA_ARGS__)
+#define LOG_ERR(...) printf(__VA_ARGS__)
+#define LOG_ALLOC_STR(str) str
+
+#include <stdio.h>
+
+#endif /* __ZEPHYR */
+
+/**
+ * Prepares JSMN parser, performs initial check of payload data and calls get/fetch/patch
+ * functions
+ */
+int ts_priv_txt_process(ts_object_t *ts);
```

Comment:
I suggest to remove the `_priv` in all functions because it makes the code more noisy. I like the idea of the dedicated `thingset_priv.h`, though.